### PR TITLE
feat!: Block reward reallocation activation at v20

### PR DESCRIPTION
**Pull request by ogabrielides**
## Issue being fixed or feature implemented
Implementation of accepted proposal: https://www.dashcentral.org/p/expedite-60-20-20-reallocation

## What was done?
Activates changers brought in #5588 on `v20` hard fork instead of `mn_rr`.

## How Has This Been Tested?
run tests

## Breaking Changes
Again, Testnet sync is broken

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: src/validation.cpp [hunk 1]
```diff
@@ -1171,35 +1171,34 @@ static std::pair<CAmount, CAmount> GetBlockSubsidyHelper(int nPrevBits, int nPre
     CAmount nSuperblockPart{};
     // Hard fork to reduce the block reward by 10 extra percent (allowing budget/superblocks)
     if (nPrevHeight > consensusParams.nBudgetPaymentsStartBlock) {
-        // Once MNRewardReallocated is active, the treasury is 20% instead of 10%
+        // Once v20 is active, the treasury is 20% instead of 10%
         // TODO remove this when we re-organize testnet
-        if (Params().NetworkIDString() == CBaseChainParams::TESTNET) fMNRewardReallocated = false;
-        nSuperblockPart = nSubsidy / (fMNRewardReallocated ? 5 : 10);
+        //if (Params().NetworkIDString() == CBaseChainParams::TESTNET) fV20Active = false;
```
Comment (ogabrielides) [inline src/validation.cpp:1176]: @UdjinM6 @PastaPastaPasta Need to decide what to do about Testnet

File: src/validation.cpp [hunk 2]
```diff
@@ -1233,8 +1232,8 @@ CAmount GetMasternodePayment(int nHeight, CAmount blockValue, bool fMNRewardReal
     }
 
     // TODO remove this when we re-organize testnet
-    if (Params().NetworkIDString() == CBaseChainParams::TESTNET) fMNRewardReallocated = false;
-    if (fMNRewardReallocated) {
+    // if (Params().NetworkIDString() == CBaseChainParams::TESTNET) fV20Active = false;
```
Comment (ogabrielides) [inline src/validation.cpp:1235]: @UdjinM6 @PastaPastaPasta Same here

Comment (UdjinM6) [reply src/validation.cpp]: we need to keep them if we want to be able to sync on testnet before #5619

Comment (UdjinM6) [reply src/validation.cpp]: also, it's much cleaner to drop them in #5619 and not here even if we want to merge them one after another imo

**Comment by UdjinM6** [comment]
pls see 5231097630

**Comment by ogabrielides** [comment]
@UdjinM6 @PastaPastaPasta Done. Please review again

**Comment by PastaPastaPasta** [comment]
syncs for me; CI looks pretty happy

**Pull request merged**